kviz/dense.py

```diff
@@ -211,6 +231,43 @@ def _make_digraph(self):
         return graph
 
 
+    def customize_graph(self, input_color='#3498db', inner_color='#2ecc71', output_color='#3498db',
+                        edge_clr='#B20000', input_shape='circle', inner_shape='circle', output_shape='circle'):
+        """
+        Changes the colors and shapes of the graph. It can be called after a DenseGraph instance is created and before
+        render().
+        It takes advantages of the existing methods _set_graph() and _make_digraph().
+
+        Parameters:
+            input_color: str.
+                    The color of the input layer in hex form (e.g. "#FFFFFF").
+                inner_color: str.
+                    The color of the inner layer(s) in hex form.
+                output_color: str.
+                    The color of the output layer in hex form.
+                edge_clr: str.
+                    The color of the edge connecting nodes of different layer. In hex form.
+                input_shape: str.
+                    The shape of the nodes in the input layer. Should be a valid shape (e.g. "polygon").
+                    Check https://graphviz.org/doc/info/shapes.html for some valid shapes.
+                    Note that now only polygon-based shapes are supported.
+                inner_shape: str.
+                    The shape of the nodes in the inner layer(s). Should be a valid shape.
+                output_shape: str.
+                    The shape of the nodes in the output layer. Should be a valid shape.
+                silence: bool.
+                    Default is False. By default, the model will be created even if some inputs are wrong; it will use
+                    default parameters instead.
+                    If set to True, an exception will raise when at least one input is incorrect.
+
+        Returns:
+            None
+        """
+        self._set_colors_shapes(input_color, inner_color, output_color,
+                                edge_clr, input_shape, inner_shape, output_shape)
+        self.set_graph(self._make_digraph())
```

Comment:
this won't work for `animate_learning`. Notice how animate learning repeatedly calls `_reset` in order to generate the animation.

Reply:
If I think animate_learning calls _snap_learning, but _snap_learning does not call any other method of this class.
The colors are determined by the code "colors = np.array([x for x in 'bgrcmyk'])". It seems that animate_learning does not call _reset?
Please tell me if I am wrong.